tests/robustness/watch.go

```diff
@@ -202,19 +203,31 @@ func validateOrdered(t *testing.T, report traffic.ClientReport) {
 	}
 }
 
-func validateUnique(t *testing.T, report traffic.ClientReport) {
-	type revisionKey struct {
-		revision int64
-		key      string
-	}
-	uniqueOperations := map[revisionKey]struct{}{}
-	for _, resp := range report.Watch {
-		for _, event := range resp.Events {
-			rk := revisionKey{key: event.Op.Key, revision: event.Revision}
-			if _, found := uniqueOperations[rk]; found {
-				t.Errorf("Broke watch guarantee: Unique - an event will never appear on a watch twice, key: %q, revision: %d, client: %d", rk.key, rk.revision, report.ClientId)
+func validateUnique(t *testing.T, expectUniqueRevision bool, report traffic.ClientReport) {
+	if expectUniqueRevision {
+		uniqueOperations := map[int64]struct{}{}
+		for _, resp := range report.Watch {
+			for _, event := range resp.Events {
+				if _, found := uniqueOperations[event.Revision]; found {
+					t.Errorf("Broke watch guarantee: Unique - an event will never appear on a watch twice, key: %q, revision: %d, client: %d", event.Op.Key, event.Revision, report.ClientId)
```

Comment:
This might be broken if we support `DeleteCollections` in robustness test. For example, `Remove all pods under a namespace`.